wpsc-includes/wpsc-countries.class.php

```diff
@@ -745,875 +768,201 @@ public static function get_country_id_by_country_code( $country_code ) {
+
+		if ( ! empty( self::$country_visibility_overrides ) ) {
+			add_action( 'wpsc_is_country_visible', array( &$this, '_get_country_visible_override' ) , 10, 2 );
+		}
+
+		self::$countries_by_id         = array();
+		self::$visible_countries_by_id = array();
+		self::$countries_by_iso_code   = array();
+
+		$countries_data_array = _wpsc_get_countries_data_array();
+
+		foreach ( $countries_data_array as $country_id => $country_data_array ) {
+			$country_data_array['id'] = $country_id;
+			$wpsc_country             = new WPSC_Country( $country_data_array );
+
+			if ( $wpsc_country->get_id() && $wpsc_country->get_isocode() ) {
+				self::$countries_by_iso_code[ $wpsc_country->get_isocode() ] = $wpsc_country;
+
+				self::$countries_by_id[ $wpsc_country->get_id() ] = $wpsc_country;
+
+				if ( $wpsc_country->is_visible() ) {
+					self::$visible_countries_by_id[ $wpsc_country->get_id() ] = $wpsc_country;
+				}
+
+				self::$currencies[ $country_data_array['currency_code'] ] = new WPSC_Currency( $country_data_array['currency_code'], $country_data_array['currency_symbol'], $country_data_array['currency_symbol_html'], $country_data_array['name'] );
+			}
+		}
+
+		$regions_data_arrays = _wpsc_get_regions_data_array();
+
+		foreach ( $regions_data_arrays as $region_id => $region_data_array ) {
+			$region_data_array['id'] = $region_id;
+			$wpsc_region             = new WPSC_Region( $region_data_array['country_id'], $region_data_array );
+
+			self::$regions_by_region_id[ $wpsc_region->get_id() ] = $wpsc_region;
+
+			if ( isset( self::$countries_by_id[ $wpsc_region->get_country_id() ] ) ) {
+				self::$countries_by_id[ $wpsc_region->get_country_id() ]->add_region( $wpsc_region );
+			}
+		}
+
+		add_action( 'wpsc_country_visibility_changed', array( &$this, 'save_country_visibility_override' ) , 10, 1 );
+	}
 
 	/**
-	 * An array that maps from country isocode to country id
-	 *
-	 * @access private
-	 * @static
-	 *
-	 * @since 3.8.14
-	 *
-	 * @var array
+	 * @param WPSC_Country $wpsc_country
 	 */
-	private static $country_id_by_iso_code = null;
+	function _get_country_visible_override( $visibility, $wpsc_country ) {
+		if ( isset( self::$country_visibility_overrides[ $wpsc_country->get_isocode() ] ) ) {
+			$visibility =  self::$country_visibility_overrides[ $wpsc_country->get_isocode() ];
+		}
+
+		return $visibility;
+	}
+
+
+	static function set_visibility( $visibility ) {
+		foreach ( self::$countries_by_id as $country_id => $wpsc_country ) {
+			$wpsc_country->set_visible( $visibility );
+		}
+	}
 
 	/**
-	 * An array that maps from country code to country id
-	 *
-	 * @access private
-	 *
-	 * @static
-	 *
-	 * @since 3.8.14
+	 * Save any cahnges to a countries visibility persistantly
 	 *
-	 * @var array
+	 * @param WPSC_Country $wpsc_country
 	 */
-	private static $country_code_by_country_id = null;
+	function save_country_visibility_override( $wpsc_country ) {
```

Comment:
Can you elaborate a bit of what's going on here, and what exactly we're storing here in the wpsc_country_visibility_overrides option?